katello/katello-backup

```diff
@@ -355,6 +355,20 @@ def backup_config_files
   puts "Done."
 end
 
+def validate_directory
+  unless postgres_write_access?
+    puts "****cancelled****"
+    puts "Postgres user needs write access to the backup directory"
+    puts "Please select a directory, such as /tmp or /var/tmp which allows Postgres write access"
+    cleanup
+  end
+end
+
+def postgres_write_access?
+  access = `sudo -u postgres test -w #{@dir} | echo $?`.strip
```

Comment:
using | to run both doesn't seem to work properly, and it'd be better to use system() to get the return status:

`system("sudo -u postgres test -w #{@dir}")`

this returns true if exit code is zero (meaning it could write), returns false if not